pkgs/development/python-modules/orange-canvas-core/default.nix

```diff
@@ -0,0 +1,79 @@
+{ lib
+, buildPythonPackage
+, pythonOlder
+, fetchFromGitHub
+, setuptools
+, setuptools-scm
+, anyqt
+, cachecontrol
+, CommonMark
+, dictdiffer
+, docutils
+, lockfile
+, qasync
+, pytestCheckHook
+}:
+
+buildPythonPackage rec {
+  pname = "orange-canvas-core";
+  version = "0.1.28";
+  format = "setuptools";
+  disabled = pythonOlder "3.6";
+
+  src = fetchFromGitHub {
+    owner = "biolab";
+    repo = pname;
+    rev = "refs/tags/${version}";
+    hash = "sha256-9ozjQQK6xJC1qf8tBWLEfnLMLH5Gdqv6ST7zwk6ox+0=";
+  };
+
+  nativeBuildInputs = [
+    setuptools
+    setuptools-scm
+  ];
+
+  propagatedBuildInputs = [
+    anyqt
+    cachecontrol
+    CommonMark
+    dictdiffer
+    docutils
+    lockfile
+    qasync
+  ];
+
+  pythonImportsCheck = [ "orangecanvas" ];
+
+  doCheck = true;
+
+  checkInputs = [
+    pytestCheckHook
+  ];
+
+   # Other tests fail because Qt modules cannot be imported
+  checkPhase = ''
+    pytest -k test_base.py
+    pytest -k test_discovery.py
+    pytest -k test_annotations.py
+    pytest -k test_signalmanager.py
+    pytest -k test_after_exit.py
+    pytest -k test_graph.py
+    pytest -k test_markup.py
+    pytest -k test_qinvoke.py
+    pytest -k test_qobjref.py
+    pytest -k test_shtools.py
+    pytest -k test_utils.py
+  '';
```

Comment:
```
  pytestFlagsArray = [
    "orangecanvas/registry/tests/test_base.py"
    "orangecanvas/registry/tests/test_discovery.py"
    "orangecanvas/scheme/tests/test_annotations.py"
    "orangecanvas/scheme/tests/test_signalmanager.py"
    "orangecanvas/utils/tests/test_after_exit.py"
    "orangecanvas/utils/tests/test_graph.py"
    "orangecanvas/utils/tests/test_markup.py"
    "orangecanvas/utils/tests/test_qinvoke.py"
    "orangecanvas/utils/tests/test_qobjref.py"
    "orangecanvas/utils/tests/test_shtools.py"
    "orangecanvas/canvas/items/tests/test_utils.py"
    "orangecanvas/utils/tests/test_utils.py"
  ];
```

Otherwise you're not really using the `pytestCheckHook`, instead you're replacing the whole `checkPhase`